radlab-ui/webapp/src/components/forms/fields/MapNestedField.tsx

```diff
@@ -0,0 +1,179 @@
+import { Field, FieldArray } from "formik"
+import { PlusIcon, XIcon } from "@heroicons/react/outline"
+import { useState } from "react"
+
+import { IUIVariable } from "@/utils/types"
+
+interface MapNestedField {
+  variable: IUIVariable
+  validate: Function
+}
+type IObjKeyPair = {
+  [key: string]: string
+}
+
+const MapNestedField: React.FC<MapNestedField> = ({ variable, validate }) => {
+  const [mapNestedParentFieldKey, setMapNestedParentFieldKey] = useState("")
+  const [mapNestedFieldKey, setMapNestedFieldKey] = useState("")
+  const [mapNestedFieldValue, setMapNestedFieldValue] = useState("")
+  const [deleteItem, setDeleteItem] = useState("")
+  const [validErrorMessage, setValidErrorMessage] = useState(null)
+  const [isValidValue, setIsValidValue] = useState(false)
+  const [isValidKey, setIsValidKey] = useState(false)
+  const [isValidParentKey, setIsValidParentKey] = useState(false)
+  const [isParentKeyEmpty, setIsParentKeyEmpty] = useState(false)
+
+  //for map nested field type
+  const handleChangeParentKey = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setMapNestedParentFieldKey(event.target.value)
+    const validationError = validate(event.target.value)
+    setValidErrorMessage(validationError)
+    validationError ? setIsValidParentKey(false) : setIsValidParentKey(true)
+  }
+
+  const handleChangeKey = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMapNestedFieldKey(event.target.value)
+    const validationError = validate(event.target.value)
+    setValidErrorMessage(validationError)
+    validationError ? setIsValidKey(false) : setIsValidKey(true)
+  }
+  const handleChangeValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMapNestedFieldValue(event.target.value)
+    const validationError = validate(event.target.value)
+    setValidErrorMessage(validationError)
+    validationError ? setIsValidValue(false) : setIsValidValue(true)
+  }
+
+  const onSetKeyValue =
+    (attributeItems: IObjKeyPair) =>
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault()
+      if (mapNestedFieldKey && mapNestedFieldValue) {
+        const checkLastKey =
+          Object.keys(attributeItems)[Object.keys(attributeItems).length - 1]
+        //@ts-ignore
```

Comment:
Instead of ignoring type checking you can use `attributeItems.hasOwnProperty(checkLastKey)`

Reply:
Changes will be made in a separate PR based on your feedback.